server/lobbyconnection.py

```diff
@@ -362,14 +362,14 @@ def send_updated_achievements(self, updated_achievements):
 
         player_id, real_username, dbPassword, steamid, create_time, ban_reason, ban_expiry = (row[i] for i in range(7))
 
+        now = datetime.now()
+        if ban_reason is not None and now < ban_expiry:
```

Comment:
Put this after password check otherwise you will be leaking ban information about all users!